haskell-src/Concordium/Types/Execution.hs

```diff
@@ -929,6 +934,24 @@ getPayload spv size = S.isolate (fromIntegral size) (S.bytesRead >>= go)
                 len <- S.getWord8
                 twswmSchedule <- replicateM (fromIntegral len) (S.get >>= \s -> S.get >>= \t -> return (s, t))
                 return TransferWithScheduleAndMemo{..}
+            25 | supportDelegation && supportSuspend -> S.label "ConfigureBaker" $ do
+                bitmap <- S.getWord16be
+                unless (bitmap .&. configureBakerBitMask == bitmap) $
+                    fail "Unsupported ConfigureBaker field(s)"
+                let maybeGet :: (S.Serialize g) => Int -> S.Get (Maybe g)
+                    maybeGet b
+                        | testBit bitmap b = Just <$> S.get
+                        | otherwise = return Nothing
+                cbCapital <- maybeGet 0
+                cbRestakeEarnings <- maybeGet 1
+                cbOpenForDelegation <- maybeGet 2
+                cbKeysWithProofs <- maybeGet 3
+                cbMetadataURL <- maybeGet 4
+                cbTransactionFeeCommission <- maybeGet 5
+                cbBakingRewardCommission <- maybeGet 6
+                cbFinalizationRewardCommission <- maybeGet 7
+                cbSuspend <- maybeGet 8
```

Comment:
There is no reason to have an entire duplicate case. Just have one case, and we don't need to test `supportSuspend` here because the definition of the bitmask determines whether `cdSuspend` can be present.